components/motor/gpio/basic.go

```diff
@@ -106,8 +107,9 @@ type Motor struct {
 	dirFlip                  bool
 	motorName                string
 
-	opMgr  operation.SingleOperationManager
-	logger golog.Logger
+	opMgr   operation.SingleOperationManager
+	logger  golog.Logger
+	stateMu sync.RWMutex
```

Comment:
See the conversation I just unresolved, I asked if I should lock mutexes as part of this PR, but was going to go forward with removing this since no one had any strong opinions.